### ISSUE TITLE
New V17 GUI is way too small in normal GUI scale, and not quite wide enough in others. 

### ISSUE THREAD

**Issue by fsendventd** [DonBruce64/MinecraftTransportSimulator#343]
From top to bottom: normal GUI scale, small HUD; normal GUI scale, proper HUD; large GUI scale, small HUD; large GUI scale, proper HUD; auto GUI scale, small HUD, auto GUI scale, proper HUD. None of them are quite right. My monitor resolution is 1920x1080, standard 16:9, if that makes any difference.
![2020-03-05_07 07 50](https://user-images.githubusercontent.com/44233145/75989455-895c3d00-5eb0-11ea-9237-ad8f83cc5830.png)
![2020-03-05_07 09 52](https://user-images.githubusercontent.com/44233145/75989557-b577be00-5eb0-11ea-81bc-a6f67b810805.png)
![2020-03-05_07 10 20](https://user-images.githubusercontent.com/44233145/75989637-d5a77d00-5eb0-11ea-872c-0d617081b779.png)
![2020-03-05_07 10 12](https://user-images.githubusercontent.com/44233145/75989650-dd672180-5eb0-11ea-9306-abca680eda80.png)
![2020-03-05_07 09 34](https://user-images.githubusercontent.com/44233145/75989660-e1933f00-5eb0-11ea-85fe-ac3e59d44ef1.png)
![2020-03-05_07 09 42](https://user-images.githubusercontent.com/44233145/75989675-e526c600-5eb0-11ea-8a33-97fdcaea1c09.png)

**Comment by DonBruce64** [comment]
This is exactly as designed.  The new GUI uses the same scale as the MC GUI.  If you've got your GUI settings set to the tiny Normal scale instead of the Auto settings, then for sure it'll be too small, as it's the same size as the inventory GUI.  I've seen the sizes y'all have for that GUI, and if you want a small inventory screen, you get a small HUD too.

As to width, it will never stretch to the edges.  Doing that would cause texture stretching and would mess up the aspect ratio of the GUI.  The whole point of adding the ability for custom GUI textures is to allow for set instrument, switch and button positions in the GUI. If I make the HUD/Panel cover the whole width, that won't apply due to differing monitor resolutions.

TL:DR It's working as intended, and I ain't changing it.

**Comment by fsendventd** [comment]
I just thought it was weird since the old one didn't do that. I can't play on any GUI scale other than normal, so I guess I'm JSONing my gauges to be optimally positioned for that on the transparent GUI...

**Comment by DonBruce64** [comment]
Do you have a picture of your inventory size in normal and large mode for reference?

**Comment by fsendventd** [comment]
Yeah, give me a minute to launch the game.

**Comment by fsendventd** [comment]
Normal first. Uses up a good amount of screen space to make things visible without making you move your mouse too far or lose awareness of what's around you. Auto second. Gigantic, takes up way too much of the screen and, due to the way I hold my mouse, actually makes me reposition my hand to move all the way across without bending my wrist too far. There's a reason normal is called that.
![2020-03-05_08 06 02](https://user-images.githubusercontent.com/44233145/75994673-8bc29500-5eb8-11ea-8dd5-a772c25640f3.png)
![2020-03-05_08 06 37](https://user-images.githubusercontent.com/44233145/75994683-8f561c00-5eb8-11ea-88dc-69de38a8a266.png)

**Comment by DonBruce64** [comment]
Yeah, the larger GUI is what I usually use as I can't see the smaller icons on my smaller screen.  Not to mention my mouse speed is rather fast, so a quick wrist flick gets me all the way across the screen.  That being said, the panel has all it's buttons, switches, and items to the same scale as the GUI, so all switches will be the same height as the arrow buttons, and all instrument item icons will be the same size as the item icons in the regular GUI.  So really it's just scaling it to match what you have set other places.

**Comment by fsendventd** [comment]
Your screen's definitely bigger than mine, remember I'm on a laptop (15.6-inch and sitting desktop distance away from it). Though now that I'm so far away from it I may have to consider making the GUI bigger...

**Comment by DonBruce64** [comment]
My screen may be larger, but the resolution is smaller.  My max resolution is 1440×900 with a 16:10 resolution.  It's kinda an odd setup in the world of 16:9 1080 setups we have nowadays, but it means I develop more for base systems via windowed mode and odd resolutions than supporting only 1080 systems.  That's one of the reasons the HUD and Panel don't extend the full width of the screen.

**Comment by fsendventd** [comment]
That's fair, my desktop's main is a 1440x900. It's what got me using normal instead of large, since large is *huge* on a lower-res screen. It just bothers me a bit that the HUD is so tiny, you can't force it to render at full size or something? I don't think anyone would be bothered by that, we'd all prefer a bigger HUD as far as I know...

**Comment by DonBruce64** [comment]
Negative.  I could force it to be larger, but then when you would change the MC GUI settings it would also change the scaling.  There's not a good way to render GUIs like HUDs and Panels size-independent of the main game without fouling up the aspect ratio or scaling.

**Comment by fsendventd** [comment]
Ah. Weird. That's unfortunate for people who use the normal GUI size...

**Comment by DonBruce64** [comment]
Well, we shall see what happens.  If enough people complain my hand will be forced.
